packages/manager/src/features/CloudPulse/shared/CloudPulseTagsFilter.test.tsx

```diff
@@ -0,0 +1,175 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+
+import { linodeFactory } from 'src/factories';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { CloudPulseTagsSelect } from './CloudPulseTagsFilter';
+
+import type { useAllLinodesQuery } from 'src/queries/linodes/linodes';
+
+const queryMocks = vi.hoisted(() => ({
+  useAllLinodesQuery: vi.fn().mockReturnValue({}),
+}));
+
+vi.mock('src/queries/linodes/linodes', async () => {
+  const actual = await vi.importActual('src/queries/linodes/linodes');
+  return {
+    ...actual,
+    useAllLinodesQuery: queryMocks.useAllLinodesQuery,
+  };
+});
+
+const mockTagsHandler = vi.fn();
+const SELECT_ALL = 'Select All';
+const ARIA_SELECTED = 'aria-selected';
+const LABEL_SUBTITLE = 'Tags (optional)';
+
+describe('CloudPulseTagsSelect component tests', () => {
+  beforeEach(() => {
+    linodeFactory.resetSequenceNumber();
+  });
+
+  it('should render a tags select component', () => {
+    queryMocks.useAllLinodesQuery.mockReturnValue({
+      data: linodeFactory.buildList(2),
+      isError: false,
+      isLoading: false,
+      status: 'success',
+    });
+    const { getByPlaceholderText, getByTestId } = renderWithTheme(
+      <CloudPulseTagsSelect
+        handleTagsChange={mockTagsHandler}
+        label="Tags"
+        resourceType={undefined}
+      />
+    );
+    expect(getByTestId('tags-select')).toBeInTheDocument();
+    expect(screen.getByLabelText(LABEL_SUBTITLE)).toBeInTheDocument();
+    expect(getByPlaceholderText('Select Tags')).toBeInTheDocument();
+  });
+
+  it('should render a Tags Select component with error message on api call failure', () => {
+    queryMocks.useAllLinodesQuery.mockReturnValue({
+      data: undefined,
+      isError: true,
+      isLoading: false,
+    } as ReturnType<typeof useAllLinodesQuery>);
+    const { getByText } = renderWithTheme(
```

Comment:
same syntax can be used for below test as well, get only the required getByText, getByRole like that and use